thorlcr/master/mawatchdog.cpp

```diff
@@ -294,7 +295,9 @@ class CMasterWatchdogUDP : public CMasterWatchdogBase
             Owned<ISocket> sock = ISocket::udp_connect(getFixedPort(masterEp.port, TPORT_watchdog), ipStr.str());
             // send empty packet, stopped set, will cease reading
             HeartBeatPacketHeader hb;
-            memset(&hb, 0, sizeof(hb));
+            //clear struct, to avoid spurious warnings when serializing raw [unpacked] struct.
+            //Cast is to avoid warning about clearing an object with non trivial copy assignment
+            memset(reinterpret_cast<void *>(&hb), 0, sizeof(hb));
```

Comment:
would it not be better to set member defaults?

Reply:
ah I see you have, so this memset to 0, isn't necessary any more I think?

Reply:
>to avoid spurious warnings when serializing raw [unpacked] struct..

Valgrind... yes I think you were right, due to alignment issues, if the whole struct isn't cleared, then there will be uninitialized data, and I think that causes false positives in valgrind.

Perhaps it would be better to pad the struct, or use a pragma to pack it..

Reply:
yes, but I think the issue it is trying to solve is to clear the bytes between the fields e.g.between a short and the next unsigned.